framework/src/problems/FEProblemBase.C

```diff
@@ -1737,6 +1737,30 @@ FEProblemBase::reinitElemFace(const Elem * elem,
     _displaced_problem->reinitElemFace(_displaced_mesh->elemPtr(elem->id()), side, bnd_id, tid);
 }
 
+void
+FEProblemBase::reinitLowerDElem(const Elem * lowerDElem, THREAD_ID tid)
+{
+  // - Set our _current_lower_d_elem for proper dof index getting in the moose variables
+  // - Reinitialize all of our lower-d FE objects so we have current phi, dphi, etc. data
+  // _assembly(tid).reinitLowerDElemRef(elem, pts, weights);
+  _assembly[tid]->reinitMortarElem(lowerDElem); // Is it?
```

Comment:
You'll still want to override `reinitLowerDElem` in `FEProblemBase` to look like this:
```c++
SubProblem::reinitLowerDElem(lower_d_elem, pts, weights, tid);

if (_displaced_problem)
  _displaced_problem->reinitLowerDElem(_displaced_mesh->elemPtr(lower_d_elem->id()), pts, weights, tid);
```
Note the `_displaced_mesh->elemPtr` code. That's critical. Otherwise you may be trying to reinit an element that doesn't even exist on the displaced mesh.

In the end product, you should have no new code in `DisplacedProblem`. Just the base implementation in `SubProblem` and an override in `FEProblemBase`

Reply:
@lindsayad thanks for the review!

Tried to follow your design suggestions. One of the tests is failing: `mortar/continuity-2d-conforming.equalgradient`. The issue appears to come from the `fe_lower->reinit(elem);` call in `Assembly::reinitLowerDElem`. An assertion in libmesh in `FE::reinit` is acting out, `libmesh_assert(this->qrule);`. Could it be that there is no default quadrature rule for that problem?

<details>
  <summary>Backtrace</summary>
1: 1   libmesh_dbg.0.dylib                 0x00000001077ebb68 libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*) + 232
2: 2   libmesh_dbg.0.dylib                 0x00000001078ceadb **libMesh::FE<1u, (libMesh::FEFamily)0>::reinit**(libMesh::Elem const*, std::__1::vector<libMesh::Point, std::__1::allocator<libMesh::Point> > const*, std::__1::vector<double, std::__1::allocator<double> > const*) + 427
3: 3   libmoose-dbg.0.dylib                0x0000000102a7f34a **Assembly::reinitLowerDElem**(libMesh::Elem const*, std::__1::vector<libMesh::Point, std::__1::allocator<libMesh::Point> > const*, std::__1::vector<double, std::__1::allocator<double> > const*) + 394
4: 4   libmoose-dbg.0.dylib                0x0000000101e57a26 **SubProblem::reinitLowerDElem**(libMesh::Elem const*, unsigned int, std::__1::vector<libMesh::Point, std::__1::allocator<libMesh::Point> > const*, std::__1::vector<double, std::__1::allocator<double> > const*) + 70
5: 5   libmoose-dbg.0.dylib                0x0000000101e57935 **FEProblemBase::reinitLowerDElem**(libMesh::Elem const*, unsigned int, std::__1::vector<libMesh::Point, std::__1::allocator<libMesh::Point> > const*, std::__1::vector<double, std::__1::allocator<double> > const*) + 69
6: 6   libmoose-dbg.0.dylib                0x00000001024cb352 **ComputeResidualThread::onBoundary(**libMesh::Elem const*, unsigned int, short, libMesh::Elem const*) + 290
7: 7   libmoose-dbg.0.dylib                0x0000000101d258ee ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> >::operator()(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, bool) + 990
8: 8   libmoose-dbg.0.dylib                0x0000000102588dec void libMesh::Threads::parallel_reduce<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>, ComputeResidualThread>(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, ComputeResidualThread&) + 156
9: 9   libmoose-dbg.0.dylib                0x0000000102581a9f NonlinearSystemBase::computeResidualInternal(std::__1::set<unsigned int, std::__1::less<unsigned int>, std::__1::allocator<unsigned int> > const&) + 863
10: 10  libmoose-dbg.0.dylib                0x00000001025811e3 NonlinearSystemBase::computeResidualTags(std::__1::set<unsigned int, std::__1::less<unsigned int>, std::__1::allocator<unsigned int> > const&) + 627
11: 11  libmoose-dbg.0.dylib                0x0000000101e7f71a FEProblemBase::computeResidualTags(std::__1::set<unsigned int, std::__1::less<unsigned int>, std::__1::allocator<unsigned int> > const&) + 2346
12: 12  libmoose-dbg.0.dylib                0x0000000101e7e866 FEProblemBase::computeResidualInternal(libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&, std::__1::set<unsigned int, std::__1::less<unsigned int>, std::__1::allocator<unsigned int> > const&) + 230
13: 13  libmoose-dbg.0.dylib                0x0000000101e7e4cc FEProblemBase::computeResidual(libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&) + 252

</details>

Reply:
```      
      QGauss qrule(elem->dim(), fe_type.default_quadrature_order());
      fe_lower->attach_quadrature_rule(&qrule);
      fe_lower->reinit(elem);
```

Maybe this will do the trick.